src/foam/nanos/ruler/RuledDAO.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.ruler',
+  name: 'RuledDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: 'Applied ruled (i.e, rule-like model) predicate on find_ and select_.',
+
+  javaImports: [
+    'foam.mlang.predicate.AbstractPredicate',
+    'foam.mlang.predicate.Predicate',
+    'foam.nanos.ruler.Ruled'
+  ],
+
+  methods: [
+    {
+      name: 'find_',
+      javaCode: `
+        var ret = getDelegate().find_(x, id);
+        if ( ret instanceof Ruled ) {
```

Comment:
I don't think this is the correct behaviour since it doesn't take into account ruleGroup or priority.
What we want is that if someone does EQ(RULE_GROUP, 'some value') then we find the the first matching item in priority order.

Reply:
This is after the find by id, priority won't have any effect, no?